In [ ]:
import pandas as pd

In [ ]:
bd = pd.read_excel('titanic3.xls')

In [ ]:
bd

pclass  survived                                             name  \
0          1         1                    Allen, Miss. Elisabeth Walton   
1          1         1                   Allison, Master. Hudson Trevor   
2          1         0                     Allison, Miss. Helen Loraine   
3          1         0             Allison, Mr. Hudson Joshua Creighton   
4          1         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)   
...      ...       ...                                              ...   
1304       3         0                             Zabour, Miss. Hileni   
1305       3         0                            Zabour, Miss. Thamine   
1306       3         0                        Zakarian, Mr. Mapriededer   
1307       3         0                              Zakarian, Mr. Ortin   
1308       3         0                               Zimmerman, Mr. Leo   

         sex      age  sibsp  parch  ticket      fare    cabin embarked boat  \
0     female  29.0000      0      0   24160  211.3375       B5        S    2   
1       male   0.9167      1      2  113781  151.5500  C22 C26        S   11   
2     female   2.0000      1      2  113781  151.5500  C22 C26        S  NaN   
3       male  30.0000      1      2  113781  151.5500  C22 C26        S  NaN   
4     female  25.0000      1      2  113781  151.5500  C22 C26        S  NaN   
...      ...      ...    ...    ...     ...       ...      ...      ...  ...   
1304  female  14.5000      1      0    2665   14.4542      NaN        C  NaN   
1305  female      NaN      1      0    2665   14.4542      NaN        C  NaN   
1306    male  26.5000      0      0    2656    7.2250      NaN        C  NaN   
1307    male  27.0000      0      0    2670    7.2250      NaN        C  NaN   
1308    male  29.0000      0      0  315082    7.8750      NaN        S  NaN   

       body                        home.dest  
0       NaN                     St Louis, MO  
1       NaN  Montreal, PQ / Chesterville, ON  
2       NaN  Montreal, PQ / Chesterville, ON  
3     135.0  Montreal, PQ / Chesterville, ON  
4       NaN  Montreal, PQ / Chesterville, ON  
...     ...                              ...  
1304  328.0                              NaN  
1305    NaN                              NaN  
1306  304.0                              NaN  
1307    NaN                              NaN  
1308    NaN                              NaN  

[1309 rows x 14 columns]

1309 Запесей

In [ ]:
bd.shape

(1309, 14)

In [ ]:
bd.isnull().sum()

pclass          0
survived        0
name            0
sex             0
age           263
sibsp           0
parch           0
ticket          0
fare            1
cabin        1014
embarked        2
boat          823
body         1188
home.dest     564
dtype: int64

In [ ]:
bd.dtypes

pclass         int64
survived       int64
name          object
sex           object
age          float64
sibsp          int64
parch          int64
ticket        object
fare         float64
cabin         object
embarked      object
boat          object
body         float64
home.dest     object
dtype: object

In [ ]:
bd.columns

Index(['pclass', 'survived', 'name', 'sex', 'age', 'sibsp', 'parch', 'ticket',
       'fare', 'cabin', 'embarked', 'boat', 'body', 'home.dest'],
      dtype='object')

In [ ]:
bd.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 14 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   pclass     1309 non-null   int64  
 1   survived   1309 non-null   int64  
 2   name       1309 non-null   object 
 3   sex        1309 non-null   object 
 4   age        1046 non-null   float64
 5   sibsp      1309 non-null   int64  
 6   parch      1309 non-null   int64  
 7   ticket     1309 non-null   object 
 8   fare       1308 non-null   float64
 9   cabin      295 non-null    object 
 10  embarked   1307 non-null   object 
 11  boat       486 non-null    object 
 12  body       121 non-null    float64
 13  home.dest  745 non-null    object 
dtypes: float64(3), int64(4), object(7)
memory usage: 143.3+ KB


In [ ]:
bd.describe()

pclass     survived          age        sibsp        parch  \
count  1309.000000  1309.000000  1046.000000  1309.000000  1309.000000   
mean      2.294882     0.381971    29.881135     0.498854     0.385027   
std       0.837836     0.486055    14.413500     1.041658     0.865560   
min       1.000000     0.000000     0.166700     0.000000     0.000000   
25%       2.000000     0.000000    21.000000     0.000000     0.000000   
50%       3.000000     0.000000    28.000000     0.000000     0.000000   
75%       3.000000     1.000000    39.000000     1.000000     0.000000   
max       3.000000     1.000000    80.000000     8.000000     9.000000   

              fare        body  
count  1308.000000  121.000000  
mean     33.295479  160.809917  
std      51.758668   97.696922  
min       0.000000    1.000000  
25%       7.895800   72.000000  
50%      14.454200  155.000000  
75%      31.275000  256.000000  
max     512.329200  328.000000

In [ ]:
bd[  bd['age']  < 18   ]

pclass  survived                                     name     sex  \
1          1         1           Allison, Master. Hudson Trevor    male   
2          1         0             Allison, Miss. Helen Loraine  female   
53         1         0                   Carrau, Mr. Jose Pedro    male   
54         1         1      Carter, Master. William Thornton II    male   
55         1         1                Carter, Miss. Lucile Polk  female   
...      ...       ...                                      ...     ...   
1265       3         0                Van Impe, Miss. Catharina  female   
1275       3         0          Vander Planke, Mr. Leo Edmondus    male   
1279       3         0     Vestrom, Miss. Hulda Amanda Adolfina  female   
1300       3         1  Yasbeck, Mrs. Antoni (Selini Alexander)  female   
1304       3         0                     Zabour, Miss. Hileni  female   

          age  sibsp  parch  ticket      fare    cabin embarked boat   body  \
1      0.9167      1      2  113781  151.5500  C22 C26        S   11    NaN   
2      2.0000      1      2  113781  151.5500  C22 C26        S  NaN    NaN   
53    17.0000      0      0  113059   47.1000      NaN        S  NaN    NaN   
54    11.0000      1      2  113760  120.0000  B96 B98        S    4    NaN   
55    14.0000      1      2  113760  120.0000  B96 B98        S    4    NaN   
...       ...    ...    ...     ...       ...      ...      ...  ...    ...   
1265  10.0000      0      2  345773   24.1500      NaN        S  NaN    NaN   
1275  16.0000      2      0  345764   18.0000      NaN        S  NaN    NaN   
1279  14.0000      0      0  350406    7.8542      NaN        S  NaN    NaN   
1300  15.0000      1      0    2659   14.4542      NaN        C  NaN    NaN   
1304  14.5000      1      0    2665   14.4542      NaN        C  NaN  328.0   

                            home.dest  
1     Montreal, PQ / Chesterville, ON  
2     Montreal, PQ / Chesterville, ON  
53                Montevideo, Uruguay  
54                      Bryn Mawr, PA  
55                      Bryn Mawr, PA  
...                               ...  
1265                              NaN  
1275                              NaN  
1279                              NaN  
1300                              NaN  
1304                              NaN  

[154 rows x 14 columns]

На корабле было 154 ребенка

In [22]:
bd[  (bd['age']  < 18)  & (bd['survived']  == 1) ].describe()

pclass  survived        age      sibsp      parch        fare  body
count  81.000000      81.0  81.000000  81.000000  81.000000   81.000000   0.0
mean    2.320988       1.0   8.013375   0.740741   1.074074   35.673200   NaN
std     0.738701       0.0   6.031362   0.905232   0.737865   44.782433   NaN
min     1.000000       1.0   0.166700   0.000000   0.000000    3.170800   NaN
25%     2.000000       1.0   2.000000   0.000000   1.000000   12.475000   NaN
50%     2.000000       1.0   7.000000   1.000000   1.000000   20.525000   NaN
75%     3.000000       1.0  14.000000   1.000000   2.000000   36.750000   NaN
max     3.000000       1.0  17.000000   4.000000   2.000000  262.375000   NaN

Выжили 81 ребенок

In [ ]:
bd[  bd['age']  < 18   ].describe()

pclass    survived         age       sibsp       parch        fare  \
count  154.000000  154.000000  154.000000  154.000000  154.000000  154.000000   
mean     2.590909    0.525974    9.101732    1.389610    1.058442   31.577219   
std      0.662868    0.500954    6.015429    1.642219    0.785797   35.818273   
min      1.000000    0.000000    0.166700    0.000000    0.000000    3.170800   
25%      2.000000    0.000000    3.000000    0.000000    0.000000   12.334375   
50%      3.000000    1.000000    9.000000    1.000000    1.000000   21.075000   
75%      3.000000    1.000000   15.750000    2.000000    2.000000   34.031250   
max      3.000000    1.000000   17.000000    8.000000    3.000000  262.375000   

             body  
count    5.000000  
mean   118.000000  
std    140.151704  
min      1.000000  
25%      4.000000  
50%     67.000000  
75%    190.000000  
max    328.000000

Возрастом от 0.16 до 17 лет. 

In [ ]:
bd[  (bd['age']  < 18) &  (bd['pclass']  == 1) ].describe()

pclass   survived        age      sibsp      parch        fare  body
count    15.0  15.000000  15.000000  15.000000  15.000000   15.000000   0.0
mean      1.0   0.866667  12.127780   0.533333   1.266667  116.062220   NaN
std       0.0   0.351866   5.889425   0.639940   0.883715   61.523141   NaN
min       1.0   0.000000   0.916700   0.000000   0.000000   39.400000   NaN
25%       1.0   1.000000   8.500000   0.000000   0.500000   69.918750   NaN
50%       1.0   1.000000  15.000000   0.000000   2.000000  110.883300   NaN
75%       1.0   1.000000  16.500000   1.000000   2.000000  143.025000   NaN
max       1.0   1.000000  17.000000   2.000000   2.000000  262.375000   NaN

пассажирами первого класса 15 детей

In [23]:
bd[  (bd['age']  < 18) &  (bd['pclass']  == 1) & (bd['survived']  == 1)].describe()

pclass  survived        age      sibsp      parch        fare  body
count    13.0      13.0  13.000000  13.000000  13.000000   13.000000   0.0
mean      1.0       1.0  12.532054   0.538462   1.307692  118.637177   NaN
std       0.0       0.0   5.455574   0.660225   0.854850   62.509993   NaN
min       1.0       1.0   0.916700   0.000000   0.000000   39.400000   NaN
25%       1.0       1.0  11.000000   0.000000   1.000000   81.858300   NaN
50%       1.0       1.0  15.000000   0.000000   2.000000  110.883300   NaN
75%       1.0       1.0  16.000000   1.000000   2.000000  134.500000   NaN
max       1.0       1.0  17.000000   2.000000   2.000000  262.375000   NaN

Спаслось 13

In [ ]:
bd[  (bd['age']  < 18) &  (bd['pclass']  == 2) ].describe()

pclass   survived        age      sibsp      parch       fare  body
count    33.0  33.000000  33.000000  33.000000  33.000000  33.000000   0.0
mean      2.0   0.878788   7.340909   0.636364   1.060606  29.704042   NaN
std       0.0   0.331434   5.909046   0.652791   0.747470  13.607017   NaN
min       2.0   0.000000   0.666700   0.000000   0.000000  10.500000   NaN
25%       2.0   1.000000   2.000000   0.000000   1.000000  23.000000   NaN
50%       2.0   1.000000   6.000000   1.000000   1.000000  26.250000   NaN
75%       2.0   1.000000  13.000000   1.000000   2.000000  37.004200   NaN
max       2.0   1.000000  17.000000   2.000000   2.000000  73.500000   NaN

Пассажирами второго класса 33 ребенка

In [24]:
bd[  (bd['age']  < 18) &  (bd['pclass']  == 2) & (bd['survived']  == 1)].describe()

pclass  survived        age      sibsp      parch       fare  body
count    29.0      29.0  29.000000  29.000000  29.000000  29.000000   0.0
mean      2.0       1.0   6.181034   0.724138   1.206897  27.766669   NaN
std       0.0       0.0   5.314143   0.648986   0.675030   9.003123   NaN
min       2.0       1.0   0.666700   0.000000   0.000000  10.500000   NaN
25%       2.0       1.0   2.000000   0.000000   1.000000  23.000000   NaN
50%       2.0       1.0   4.000000   1.000000   1.000000  26.250000   NaN
75%       2.0       1.0   8.000000   1.000000   2.000000  36.750000   NaN
max       2.0       1.0  17.000000   2.000000   2.000000  41.579200   NaN

Спаслось 29

In [ ]:
bd[  (bd['age']  < 18) &  (bd['pclass']  == 3) ].describe()

pclass    survived         age       sibsp       parch        fare  \
count   106.0  106.000000  106.000000  106.000000  106.000000  106.000000   
mean      3.0    0.367925    9.221698    1.745283    1.028302   20.204953   
std       0.0    0.484532    5.929936    1.826062    0.786282   12.529823   
min       3.0    0.000000    0.166700    0.000000    0.000000    3.170800   
25%       3.0    0.000000    4.000000    0.000000    0.000000    9.256250   
50%       3.0    0.000000    9.000000    1.000000    1.000000   16.000000   
75%       3.0    1.000000   15.750000    4.000000    2.000000   29.125000   
max       3.0    1.000000   17.000000    8.000000    3.000000   69.550000   

             body  
count    5.000000  
mean   118.000000  
std    140.151704  
min      1.000000  
25%      4.000000  
50%     67.000000  
75%    190.000000  
max    328.000000

Пассажирами третьего класса 106 ребенка

In [25]:
bd[  (bd['age']  < 18) &  (bd['pclass']  == 3) & (bd['survived']  == 1)].describe()

pclass  survived        age      sibsp      parch       fare  body
count    39.0      39.0  39.000000  39.000000  39.000000  39.000000   0.0
mean      3.0       1.0   7.869659   0.820513   0.897436  13.897756   NaN
std       0.0       0.0   6.066932   1.120897   0.717997   6.268020   NaN
min       3.0       1.0   0.166700   0.000000   0.000000   3.170800   NaN
25%       3.0       1.0   2.500000   0.000000   0.000000   8.516700   NaN
50%       3.0       1.0   6.000000   1.000000   1.000000  12.475000   NaN
75%       3.0       1.0  14.500000   1.000000   1.000000  16.700000   NaN
max       3.0       1.0  17.000000   4.000000   2.000000  31.387500   NaN

Спаслось 39

Дети 1 класса спаслись 86%
Дети 2 класса спаслись 87%
Дети 3 класса спаслись 36%

In [20]:
bd[  (bd['age']  < 18)  & (bd['survived']  == 1) ].describe()

pclass  survived        age      sibsp      parch        fare  body
count  81.000000      81.0  81.000000  81.000000  81.000000   81.000000   0.0
mean    2.320988       1.0   8.013375   0.740741   1.074074   35.673200   NaN
std     0.738701       0.0   6.031362   0.905232   0.737865   44.782433   NaN
min     1.000000       1.0   0.166700   0.000000   0.000000    3.170800   NaN
25%     2.000000       1.0   2.000000   0.000000   1.000000   12.475000   NaN
50%     2.000000       1.0   7.000000   1.000000   1.000000   20.525000   NaN
75%     3.000000       1.0  14.000000   1.000000   2.000000   36.750000   NaN
max     3.000000       1.0  17.000000   4.000000   2.000000  262.375000   NaN

Спаслись 81 ребенок

In [21]:
bd[  bd['age']  > 17].describe()

pclass    survived         age       sibsp       parch        fare  \
count  892.000000  892.000000  892.000000  892.000000  892.000000  891.000000   
mean     2.141256    0.387892   33.468610    0.349776    0.310538   37.569093   
std      0.851619    0.487543   12.244544    0.594466    0.799116   58.461348   
min      1.000000    0.000000   18.000000    0.000000    0.000000    0.000000   
25%      1.000000    0.000000   24.000000    0.000000    0.000000    8.050000   
50%      2.000000    0.000000   30.000000    0.000000    0.000000   14.454200   
75%      3.000000    1.000000   41.000000    1.000000    0.000000   36.125000   
max      3.000000    1.000000   80.000000    4.000000    6.000000  512.329200   

             body  
count  115.000000  
mean   162.739130  
std     96.297245  
min      7.000000  
25%     77.000000  
50%    156.000000  
75%    257.000000  
max    327.000000

На корабле 892 взрослых, 154 ребенка и 263 человека у которых не указан возраст

In [27]:
bd[  bd['age']  > 17].describe()

pclass    survived         age       sibsp       parch        fare  \
count  892.000000  892.000000  892.000000  892.000000  892.000000  891.000000   
mean     2.141256    0.387892   33.468610    0.349776    0.310538   37.569093   
std      0.851619    0.487543   12.244544    0.594466    0.799116   58.461348   
min      1.000000    0.000000   18.000000    0.000000    0.000000    0.000000   
25%      1.000000    0.000000   24.000000    0.000000    0.000000    8.050000   
50%      2.000000    0.000000   30.000000    0.000000    0.000000   14.454200   
75%      3.000000    1.000000   41.000000    1.000000    0.000000   36.125000   
max      3.000000    1.000000   80.000000    4.000000    6.000000  512.329200   

             body  
count  115.000000  
mean   162.739130  
std     96.297245  
min      7.000000  
25%     77.000000  
50%    156.000000  
75%    257.000000  
max    327.000000

Взрослых было 892 человека

In [33]:
bd[  (bd['sex']  == 'male') & (bd['age']  > 17)].describe()

pclass    survived         age       sibsp       parch        fare  \
count  576.000000  576.000000  576.000000  576.000000  576.000000  575.000000   
mean     2.251736    0.180556   33.632812    0.270833    0.166667   28.055607   
std      0.830399    0.384984   12.375016    0.530381    0.598549   46.753588   
min      1.000000    0.000000   18.000000    0.000000    0.000000    0.000000   
25%      1.750000    0.000000   24.000000    0.000000    0.000000    7.895800   
50%      3.000000    0.000000   30.000000    0.000000    0.000000   13.000000   
75%      3.000000    0.000000   41.000000    0.000000    0.000000   26.550000   
max      3.000000    1.000000   80.000000    3.000000    6.000000  512.329200   

             body  
count  108.000000  
mean   163.981481  
std     94.257649  
min      9.000000  
25%     80.750000  
50%    160.500000  
75%    256.500000  
max    322.000000

576 мужчин

In [35]:
bd[  (bd['sex']  == 'female') & (bd['age']  > 17)].describe()

pclass    survived         age       sibsp       parch        fare  \
count  316.000000  316.000000  316.000000  316.000000  316.000000  316.000000   
mean     1.939873    0.765823   33.169304    0.493671    0.572785   54.880025   
std      0.854229    0.424155   12.016747    0.673742    1.022487   72.162452   
min      1.000000    0.000000   18.000000    0.000000    0.000000    6.750000   
25%      1.000000    1.000000   23.750000    0.000000    0.000000   13.000000   
50%      2.000000    1.000000   30.000000    0.000000    0.000000   26.000000   
75%      3.000000    1.000000   40.000000    1.000000    1.000000   71.070825   
max      3.000000    1.000000   76.000000    4.000000    6.000000  512.329200   

             body  
count    7.000000  
mean   143.571429  
std    131.503938  
min      7.000000  
25%     52.500000  
50%     61.000000  
75%    252.500000  
max    327.000000

316 женщин

In [36]:
bd[  (bd['sex']  == 'male') & (bd['age']  > 17) & (bd['pclass']  == 1)].describe()

pclass    survived         age       sibsp       parch        fare  \
count   144.0  144.000000  144.000000  144.000000  144.000000  144.000000   
mean      1.0    0.326389   42.545139    0.395833    0.263889   72.147802   
std       0.0    0.470528   13.089246    0.557451    0.625631   76.099156   
min       1.0    0.000000   18.000000    0.000000    0.000000    0.000000   
25%       1.0    0.000000   31.000000    0.000000    0.000000   28.500000   
50%       1.0    0.000000   42.000000    0.000000    0.000000   52.000000   
75%       1.0    1.000000   51.000000    1.000000    0.000000   79.312500   
max       1.0    1.000000   80.000000    3.000000    4.000000  512.329200   

             body  
count   35.000000  
mean   162.828571  
std     82.652172  
min     16.000000  
25%    109.500000  
50%    166.000000  
75%    233.000000  
max    307.000000

144 Мужчин пассажиры 1 класса

In [37]:
bd[  (bd['sex']  == 'male') & (bd['age']  > 17) & (bd['pclass']  == 1)& (bd['survived']  == 1)].describe()

pclass  survived        age      sibsp      parch        fare  body
count    47.0      47.0  47.000000  47.000000  47.000000   47.000000   0.0
mean      1.0       1.0  39.680851   0.446809   0.191489   70.846638   NaN
std       0.0       0.0  11.922041   0.544079   0.449071   98.732329   NaN
min       1.0       1.0  21.000000   0.000000   0.000000    0.000000   NaN
25%       1.0       1.0  30.500000   0.000000   0.000000   28.110400   NaN
50%       1.0       1.0  37.000000   0.000000   0.000000   52.000000   NaN
75%       1.0       1.0  48.500000   1.000000   0.000000   76.729200   NaN
max       1.0       1.0  80.000000   2.000000   2.000000  512.329200   NaN

Спаслось 47 это 32%

In [38]:
bd[  (bd['sex']  == 'female') & (bd['age']  > 17) & (bd['pclass']  == 1)].describe()

pclass    survived         age       sibsp       parch        fare  \
count   125.0  125.000000  125.000000  125.000000  125.000000  125.000000   
mean      1.0    0.968000   38.504000    0.560000    0.480000  112.503368   
std       0.0    0.176708   13.393923    0.688617    0.789038   86.686821   
min       1.0    0.000000   18.000000    0.000000    0.000000   25.700000   
25%       1.0    1.000000   27.000000    0.000000    0.000000   55.900000   
50%       1.0    1.000000   36.000000    0.000000    0.000000   80.000000   
75%       1.0    1.000000   49.000000    1.000000    1.000000  146.520800   
max       1.0    1.000000   76.000000    3.000000    4.000000  512.329200   

       body  
count   0.0  
mean    NaN  
std     NaN  
min     NaN  
25%     NaN  
50%     NaN  
75%     NaN  
max     NaN

125 женщин пассажиры 1 класса

In [39]:
bd[  (bd['sex']  == 'female') & (bd['age']  > 17) & (bd['pclass']  == 1)& (bd['survived']  == 1)].describe()

pclass  survived         age       sibsp       parch        fare  body
count   121.0     121.0  121.000000  121.000000  121.000000  121.000000   0.0
mean      1.0       1.0   38.338843    0.561983    0.479339  112.638017   NaN
std       0.0       0.0   13.329512    0.693933    0.786333   86.832670   NaN
min       1.0       1.0   18.000000    0.000000    0.000000   25.700000   NaN
25%       1.0       1.0   27.000000    0.000000    0.000000   56.929200   NaN
50%       1.0       1.0   36.000000    0.000000    0.000000   80.000000   NaN
75%       1.0       1.0   48.000000    1.000000    1.000000  136.779200   NaN
max       1.0       1.0   76.000000    3.000000    4.000000  512.329200   NaN

Спаслось 39 это 31%

In [40]:
bd[  (bd['sex']  == 'male') & (bd['age']  > 17) & (bd['pclass']  == 2)].describe()

pclass    survived         age       sibsp       parch        fare  \
count   143.0  143.000000  143.000000  143.000000  143.000000  143.000000   
mean      2.0    0.083916   33.395105    0.328671    0.132867   19.799795   
std       0.0    0.278236   11.889878    0.553812    0.415166   14.072117   
min       2.0    0.000000   18.000000    0.000000    0.000000    9.687500   
25%       2.0    0.000000   25.000000    0.000000    0.000000   12.437500   
50%       2.0    0.000000   30.000000    0.000000    0.000000   13.000000   
75%       2.0    0.000000   40.000000    1.000000    0.000000   26.000000   
max       2.0    1.000000   70.000000    2.000000    2.000000   73.500000   

             body  
count   30.000000  
mean   171.233333  
std    106.708007  
min     15.000000  
25%     85.000000  
50%    160.000000  
75%    285.250000  
max    322.000000

143 Мужчин пассажиры 2 класса

In [41]:
bd[  (bd['sex']  == 'male') & (bd['age']  > 17) & (bd['pclass']  == 2)& (bd['survived']  == 1)].describe()

pclass  survived        age      sibsp      parch       fare  body
count    12.0      12.0  12.000000  12.000000  12.000000  12.000000   0.0
mean      2.0       1.0  30.916667   0.166667   0.083333  14.704858   NaN
std       0.0       0.0  11.743148   0.389249   0.288675   6.154606   NaN
min       2.0       1.0  19.000000   0.000000   0.000000  10.500000   NaN
25%       2.0       1.0  23.500000   0.000000   0.000000  10.500000   NaN
50%       2.0       1.0  29.500000   0.000000   0.000000  13.000000   NaN
75%       2.0       1.0  32.500000   0.000000   0.000000  13.859350   NaN
max       2.0       1.0  62.000000   1.000000   1.000000  29.000000   NaN

Спаслось 12 это 8%

In [42]:
bd[  (bd['sex']  == 'female') & (bd['age']  > 17) & (bd['pclass']  == 2)].describe()

pclass   survived        age      sibsp      parch       fare  body
count    85.0  85.000000  85.000000  85.000000  85.000000  85.000000   1.0
mean      2.0   0.870588  31.570588   0.482353   0.564706  22.265441  52.0
std       0.0   0.337647  10.019966   0.628902   0.851428  11.429334   NaN
min       2.0   0.000000  18.000000   0.000000   0.000000  10.500000  52.0
25%       2.0   1.000000  24.000000   0.000000   0.000000  13.000000  52.0
50%       2.0   1.000000  29.000000   0.000000   0.000000  21.000000  52.0
75%       2.0   1.000000  36.000000   1.000000   1.000000  26.000000  52.0
max       2.0   1.000000  60.000000   3.000000   3.000000  65.000000  52.0

85 женщин пассажиры 2 класса

In [43]:
bd[  (bd['sex']  == 'female') & (bd['age']  > 17) & (bd['pclass']  == 2)& (bd['survived']  == 1)].describe()

pclass  survived        age      sibsp      parch       fare  body
count    74.0      74.0  74.000000  74.000000  74.000000  74.000000   0.0
mean      2.0       1.0  31.195946   0.472973   0.621622  22.757601   NaN
std       0.0       0.0   9.347484   0.645808   0.886778  11.944024   NaN
min       2.0       1.0  18.000000   0.000000   0.000000  10.500000   NaN
25%       2.0       1.0  24.000000   0.000000   0.000000  13.000000   NaN
50%       2.0       1.0  29.000000   0.000000   0.000000  22.000000   NaN
75%       2.0       1.0  36.000000   1.000000   1.000000  26.000000   NaN
max       2.0       1.0  55.000000   3.000000   3.000000  65.000000   NaN

Спаслось 74 это 87%

In [44]:
bd[  (bd['sex']  == 'male') & (bd['age']  > 17) & (bd['pclass']  == 3)].describe()

pclass    survived         age       sibsp       parch        fare  \
count   289.0  289.000000  289.000000  289.000000  289.000000  288.000000   
mean      3.0    0.155709   29.309689    0.179931    0.134948   10.108750   
std       0.0    0.363209    9.616705    0.488207    0.655471    7.466379   
min       3.0    0.000000   18.000000    0.000000    0.000000    0.000000   
25%       3.0    0.000000   22.000000    0.000000    0.000000    7.747925   
50%       3.0    0.000000   27.000000    0.000000    0.000000    7.895800   
75%       3.0    0.000000   34.000000    0.000000    0.000000    8.662500   
max       3.0    1.000000   74.000000    3.000000    6.000000   56.495800   

             body  
count   43.000000  
mean   159.860465  
std     95.843270  
min      9.000000  
25%     71.000000  
50%    156.000000  
75%    257.000000  
max    314.000000

289 Мужчин пассажиры 3 класса

In [45]:
bd[  (bd['sex']  == 'male') & (bd['age']  > 17) & (bd['pclass']  == 3)& (bd['survived']  == 1)].describe()

pclass  survived        age      sibsp      parch       fare  body
count    45.0      45.0  45.000000  45.000000  45.000000  45.000000   0.0
mean      3.0       1.0  27.322222   0.155556   0.066667  12.021664   NaN
std       0.0       0.0   6.393494   0.520295   0.252262  12.518043   NaN
min       3.0       1.0  18.000000   0.000000   0.000000   0.000000   NaN
25%       3.0       1.0  22.000000   0.000000   0.000000   7.775000   NaN
50%       3.0       1.0  26.000000   0.000000   0.000000   7.895800   NaN
75%       3.0       1.0  32.000000   0.000000   0.000000   8.662500   NaN
max       3.0       1.0  45.000000   3.000000   1.000000  56.495800   NaN

Спаслось 45 это 15%

In [46]:
bd[  (bd['sex']  == 'female') & (bd['age']  > 17) & (bd['pclass']  == 3)].describe()

pclass    survived         age       sibsp       parch        fare  \
count   106.0  106.000000  106.000000  106.000000  106.000000  106.000000   
mean      3.0    0.443396   28.160377    0.424528    0.688679   13.081174   
std       0.0    0.499146    8.930298    0.689349    1.340669    8.097223   
min       3.0    0.000000   18.000000    0.000000    0.000000    6.750000   
25%       3.0    0.000000   22.000000    0.000000    0.000000    7.775000   
50%       3.0    0.000000   26.000000    0.000000    0.000000    9.156250   
75%       3.0    1.000000   32.750000    1.000000    1.000000   15.822925   
max       3.0    1.000000   63.000000    4.000000    6.000000   46.900000   

             body  
count    6.000000  
mean   158.833333  
std    137.096195  
min      7.000000  
25%     55.000000  
50%    133.500000  
75%    275.750000  
max    327.000000

106 женщин пассажиры 3 класса

In [48]:
bd[  (bd['sex']  == 'female') & (bd['age']  > 17) & (bd['pclass']  == 3)& (bd['survived']  == 1)].describe()

pclass  survived        age      sibsp      parch       fare  body
count    47.0      47.0  47.000000  47.000000  47.000000  47.000000   0.0
mean      3.0       1.0  27.361702   0.340426   0.531915  11.595923   NaN
std       0.0       0.0   9.246446   0.599877   1.018333   5.407092   NaN
min       3.0       1.0  18.000000   0.000000   0.000000   6.950000   NaN
25%       3.0       1.0  22.000000   0.000000   0.000000   7.750000   NaN
50%       3.0       1.0  24.000000   0.000000   0.000000   8.683300   NaN
75%       3.0       1.0  31.000000   1.000000   1.000000  15.493750   NaN
max       3.0       1.0  63.000000   3.000000   5.000000  31.387500   NaN

Спаслось 47 это 44%

In [55]:
bd[ bd['pclass']  == 3].describe()

pclass    survived         age       sibsp       parch        fare  \
count   709.0  709.000000  501.000000  709.000000  709.000000  708.000000   
mean      3.0    0.255289   24.816367    0.568406    0.400564   13.302889   
std       0.0    0.436331   11.958202    1.299681    0.981639   11.494358   
min       3.0    0.000000    0.166700    0.000000    0.000000    0.000000   
25%       3.0    0.000000   18.000000    0.000000    0.000000    7.750000   
50%       3.0    0.000000   24.000000    0.000000    0.000000    8.050000   
75%       3.0    1.000000   32.000000    1.000000    0.000000   15.245800   
max       3.0    1.000000   74.000000    8.000000    9.000000   69.550000   

             body  
count   55.000000  
mean   155.818182  
std    102.403720  
min      1.000000  
25%     67.500000  
50%    153.000000  
75%    257.000000  
max    328.000000

Транспортные расходы 3 класса. До 69 долларов, средняя 8 долларов, медианная 13 долларов

In [56]:
bd[ bd['pclass']  == 2].describe()

pclass    survived         age       sibsp       parch        fare  \
count   277.0  277.000000  261.000000  277.000000  277.000000  277.000000   
mean      2.0    0.429603   29.506705    0.393502    0.368231   21.179196   
std       0.0    0.495915   13.638628    0.590100    0.692717   13.607122   
min       2.0    0.000000    0.666700    0.000000    0.000000    0.000000   
25%       2.0    0.000000   22.000000    0.000000    0.000000   13.000000   
50%       2.0    0.000000   29.000000    0.000000    0.000000   15.045800   
75%       2.0    1.000000   36.000000    1.000000    1.000000   26.000000   
max       2.0    1.000000   70.000000    3.000000    3.000000   73.500000   

             body  
count   31.000000  
mean   167.387097  
std    107.077753  
min     15.000000  
25%     78.000000  
50%    155.000000  
75%    284.500000  
max    322.000000

Транспортные расходы 2 класса. До 73 долларов, средняя 15 долларов, медианная 21 долларов

In [57]:
bd[ bd['pclass']  == 1].describe()

pclass    survived         age       sibsp       parch        fare  \
count   323.0  323.000000  284.000000  323.000000  323.000000  323.000000   
mean      1.0    0.619195   39.159918    0.436533    0.365325   87.508992   
std       0.0    0.486338   14.548059    0.609064    0.715602   80.447178   
min       1.0    0.000000    0.916700    0.000000    0.000000    0.000000   
25%       1.0    0.000000   28.000000    0.000000    0.000000   30.695800   
50%       1.0    1.000000   39.000000    0.000000    0.000000   60.000000   
75%       1.0    1.000000   50.000000    1.000000    0.500000  107.662500   
max       1.0    1.000000   80.000000    3.000000    4.000000  512.329200   

             body  
count   35.000000  
mean   162.828571  
std     82.652172  
min     16.000000  
25%    109.500000  
50%    166.000000  
75%    233.000000  
max    307.000000

Транспортные расходы 1 класса. До 512 долларов, средняя 60 долларов, медианная 87 долларов